lib/libspl/include/sys/dkio.h

```diff
@@ -476,6 +522,37 @@ typedef struct dk_updatefw_32 {
 #define	FW_TYPE_TEMP	0x0		/* temporary use */
 #define	FW_TYPE_PERM	0x1		/* permanent use */
 
+/*
+ * ioctl to free space (e.g. SCSI UNMAP) off a disk.
+ * Pass a dkioc_free_list_t containing a list of extents to be freed.
+ */
+#define	DKIOCFREE	(DKIOC|50)
+
+#define	DF_WAIT_SYNC	0x00000001	/* Wait for full write-out of free. */
```

Comment:
Preserved from illumos, just in case we need it.  Now that you mention it we could probably do something like pass `DF_WAIT_SYNC` when `zfs_trim_sync` is set rather than checking directly for `zfs_trim_sync` in `vdev_disk.c`.